specifyweb/frontend/js_src/lib/components/Router/ReactLazy.tsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+
+import type { IR } from '../../utils/types';
+import { LoadingScreen } from '../Molecules/Dialog';
+
+function FakeAsync<PROPS extends IR<unknown>>({
+  Element,
+  props,
+}: {
+  readonly Element: React.FunctionComponent<PROPS>;
+  readonly props: PROPS;
+}) {
+  return (
+    <React.Suspense fallback={<LoadingScreen />}>
+      <Element {...props} />
+    </React.Suspense>
+  );
+}
+
+export function ReactLazy<PROPS extends IR<unknown>>(
```

Comment:
can you add this comment to ReactLazy?

```ts
/**
 * Using this allows Webpack to split code bundles.
 * React Suspense takes care of rendering a loading screen if component is
 * being fetched.
 * Having a separate Suspense for each async component rather than a one main
 * suspense on the top level prevents all components from being un-rendered
 * when any component is being loaded.
 */
```

Reply:
(comment coppied from the comment Async function had)

and lol, reading the comment I see that I mentioned the exact thing that caused #4308 to happen 🤣
> * Having a separate Suspense for each async component rather than a one main
> * suspense on the top level prevents all components from being un-rendered
> * when any component is being loaded.

my react skills are getting rusty🤣 (we don't use React at esri - but we have a Frankenstein called Esriact - less said about it, the better)

Reply:
I thought that React would be smart and try to find the first parent Suspense (which would have not caused https://github.com/specify/specify7/issues/4308) , but in my experimentation it doesn't do that, so React.lazy needs to be immediately parented by the Suspense!